pkg/kubelet/status/manager_test.go

```diff
@@ -714,3 +714,58 @@ func expectPodStatus(t *testing.T, m *manager, pod *api.Pod) api.PodStatus {
 	}
 	return status
 }
+
+func TestDeletePods(t *testing.T) {
+	pod := getTestPod()
+	// Set the deletion timestamp.
+	pod.DeletionTimestamp = new(unversioned.Time)
+	client := testclient.NewSimpleFake(pod)
+	m := newTestManager(client)
+	m.podManager.AddPod(pod)
+
+	status := getRandomPodStatus()
+	now := unversioned.Now()
+	status.StartTime = &now
+	m.SetPodStatus(pod, status)
+
+	m.testSyncBatch()
+	// Expect to see an delete action.
+	verifyActions(t, m.kubeClient, []testclient.Action{
+		testclient.GetActionImpl{ActionImpl: testclient.ActionImpl{Verb: "get", Resource: "pods"}},
+		testclient.UpdateActionImpl{ActionImpl: testclient.ActionImpl{Verb: "update", Resource: "pods", Subresource: "status"}},
+		testclient.DeleteActionImpl{ActionImpl: testclient.ActionImpl{Verb: "delete", Resource: "pods"}},
+	})
+}
+
+func TestDoNotDeleteMirrorPods(t *testing.T) {
+	staticPod := getTestPod()
+	staticPod.Annotations = map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"}
+	mirrorPod := getTestPod()
+	mirrorPod.UID = "mirror-12345678"
+	mirrorPod.Annotations = map[string]string{
+		kubetypes.ConfigSourceAnnotationKey: "api",
+		kubetypes.ConfigMirrorAnnotationKey: "mirror",
+	}
+	// Set the deletion timestamp.
+	mirrorPod.DeletionTimestamp = new(unversioned.Time)
+	client := testclient.NewSimpleFake(mirrorPod)
+	m := newTestManager(client)
+	m.podManager.AddPod(staticPod)
+	m.podManager.AddPod(mirrorPod)
+	// Verify setup.
+	assert.True(t, kubepod.IsStaticPod(staticPod), "SetUp error: staticPod")
+	assert.True(t, kubepod.IsMirrorPod(mirrorPod), "SetUp error: mirrorPod")
+	assert.Equal(t, m.podManager.TranslatePodUID(mirrorPod.UID), staticPod.UID)
+
+	status := getRandomPodStatus()
+	now := unversioned.Now()
+	status.StartTime = &now
+	m.SetPodStatus(staticPod, status)
```

Comment:
I think the start time was here so that `SetPodStatus()` doesn't record a start time itself. If you want to compare the expected and the actual status, you'll need that.